[caffe2] Move CAFFE SDT macros' definitions to `c10/util/`

vlad-scherbich opened the pull request:
Summary: Moving static tracepoint macros header to a location where it can be easily used by various PyTorch components (`c10/utill`).

Test Plan:
Same as for D47159249:

Tested the following macros on test scripts with libbpf USDTs:
* `CAFFE_SDT`
* `CAFFE_DISABLE_SDT`
* `CAFFE_SDT_WITH_SEMAPHORE`

Reviewed By: EDG-GH

Differential Revision: D47636258

EDG-GH reviewed (approved):
Originally reviewed and accepted internally.

vlad-scherbich committed:
[caffe2] Move CAFFE SDT macros' definitions to `c10/util/` (#105856)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/105856

Moving static tracepoint macros header to a location where it can be easily used by various PyTorch components (`c10/utill`).

Test Plan:
Same as for D47159249:

Tested the following macros on test scripts with libbpf USDTs:
* `CAFFE_SDT`
* `CAFFE_DISABLE_SDT`
* `CAFFE_SDT_WITH_SEMAPHORE`

Reviewed By: chaekit, EDG-GH

Differential Revision: D47636258

fbshipit-source-id: 6ce33f86db2909a07092213a05d04448fc7041d4

v-s-2 commented:
/easycla
